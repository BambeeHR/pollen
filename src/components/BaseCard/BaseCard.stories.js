import { boolean } from '@storybook/addon-knobs';
import BaseCard from './BaseCard.vue';

export default {
  title: 'BaseCard',
  component: BaseCard,
  parameters: {
    componentSubtitle: 'Basic card component.',
  },
};

export const WithKnobs = () => {
  return {
    components: { BaseCard },
    props: {
      fullBleed: {
        default: boolean('Full-bleed', false),
      },
    },
    template: `
      <BaseCard
        :full-bleed="fullBleed"
      >
        <template #header>
          <div class="text-center font-body-large font-bold pt-3">
            This is a header!
          </div>
        </template>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod 
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim 
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea 
        commodo consequat. Duis aute irure dolor in reprehenderit in voluptate 
        velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint 
        occaecat cupidatat non proident, sunt in culpa qui officia deserunt 
        mollit anim id est laborum.
        <template #footer>
          This is a footer.
        </template>
      </BaseCard>
    `,
  };
};

/* eslint-disable no-param-reassign */
[WithKnobs].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['BaseCard.test.js'],
    },
  };
});
