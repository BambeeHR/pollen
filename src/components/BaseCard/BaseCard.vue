<template functional>
  <component
    :is="props.tag"
    class="base-card"
    :class="[
      { 'base-card--full-bleed': props.fullBleed },
      data.class,
      data.staticClass,
    ]"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  >
    <template v-if="$slots.header">
      <header class="base-card__header">
        <slot name="header" />
      </header>
    </template>
    <section class="base-card__content">
      <slot />
    </section>
    <template v-if="$slots.footer">
      <footer class="base-card__footer">
        <slot name="footer" />
      </footer>
    </template>
  </component>
</template>

<script>
import getAttributes from '../../util/getAttributes';

export default {
  getAttributes,
  props: {
    /** If true, removes the internal padding of the card. */
    fullBleed: {
      type: Boolean,
      default: false,
    },
    /** Defines the semantic tag for the root element. */
    tag: {
      type: String,
      default: 'article',
    },
  },
};
</script>

<style scoped>
.base-card {
  @apply bg-white flex flex-col rounded-md shadow-1 w-full;
}

.base-card__content {
  @apply flex-grow px-6 py-4;
}

.base-card__footer {
  @apply bg-gray-7 font-body-caption p-3 text-center text-gray-3;
}

.base-card--full-bleed .base-card__content {
  @apply p-0;
}
</style>
