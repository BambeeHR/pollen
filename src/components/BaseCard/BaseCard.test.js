import { mount } from '@vue/test-utils';
import BaseCard from './BaseCard.vue';

describe('BaseCard', () => {
  test('renders correctly', () => {
    const wrapper = mount(BaseCard, {
      slots: {
        header: 'This is the header',
        default: 'Hello world',
        footer: 'This is the footer',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
