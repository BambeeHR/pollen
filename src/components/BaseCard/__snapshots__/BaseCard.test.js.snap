// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard renders correctly 1`] = `
<article
  class="base-card"
>
  <header
    class="base-card__header"
  >
    This is the header
  </header>
   
  <section
    class="base-card__content"
  >
    Hello world
  </section>
   
  <footer
    class="base-card__footer"
  >
    This is the footer
  </footer>
</article>
`;
